### PR TITLE
Add virus-scan action

### PR DESCRIPTION
**Pull request by eirascon**
Signed-off-by: eirascon <eduardo.i.rascon.garcia@intel.com>

### CODE REVIEW

File: virus-scan/action.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    description: URL to get mcAfee updates
+    default: https://update.nai.com/products/commonupdater
+    required: true
+  scan-directory:
+    description: Directory name where the files to be scanned are grouped
+    default: scanme
+    required: true
+  scan-results:
+    description: File where the scan results are stored
+    default: scan_results.txt
```
Comment (garciacu) [inline virus-scan/action.yml:15]: It's easier if the you build the name of the file here than if it's passed over here maybe user can just pass the name of the repo and you can complete the file name when this action is executed

**Review by garciacu — CHANGES_REQUESTED** [review]

File: virus-scan/action.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    required: true
+  target:
+    description: Comma separated list of file extensions to be scanned
+    required: true
+  key:
+    description: Repository name followed by job name and commit-sha
```
Comment (garciacu) [inline virus-scan/action.yml:21]: It's easier if the you build the name of the file here than if it's passed over here maybe user can just pass the name of the repo and you can complete the file name when this action is executed

**Review by garciacu — CHANGES_REQUESTED** [review]

**Review by garciacu — APPROVED** [review]

**Review by garciacu — DISMISSED** [review]

File: virus-scan/action.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+name: "Virus Scan"
+description: "McAfee virus scanner"
+
+inputs:
+  commonupdater-url:
+    description: URL to get mcAfee updates
+    default: https://update.nai.com/products/commonupdater
+    required: true
```
Comment (juliocri) [inline virus-scan/action.yml:8]: This input can be optional

File: virus-scan/action.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  scan-directory:
+    description: Directory name where the files to be scanned are grouped
+    default: scanme
+    required: true
```
Comment (juliocri) [inline virus-scan/action.yml:12]: This input can be optional

File: virus-scan/action.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  scan-results:
+    description: File where the scan results are stored
+    default: scan_results.txt
+    required: true
```
Comment (juliocri) [inline virus-scan/action.yml:16]: This input can be optional

File: virus-scan/action.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Create directory and group files to be scanned
+      shell: sh
+      env:
+        FILE_EXT: ${{ inputs.target }}
+      run: |
+        mkdir ${{ inputs.scan-directory }}
+        for i in $(echo $FILE_EXT | sed "s/,/ /g"); do
+          find . -type f -name *."$i" -exec mv '{}' ${{ inputs.scan-directory }}/  \;
```
Comment (juliocri) [inline virus-scan/action.yml:33]: The for loop can be avoided if we sen the `input.target` in a format like: `*.{dll,pdb,cs,txt}`

**Review by juliocri — CHANGES_REQUESTED** [review]

File: virus-scan/action.yml (hunk 6 above)
Comment (eirascon) [reply virus-scan/action.yml:33]: I have been working on this but I'm having issues to implement the suggested change.  Do you have an explicit example on how to do this?

**Review by garciacu — APPROVED** [review]

**Pull request merged**